hbase-common/src/main/java/org/apache/hadoop/hbase/util/Bytes.java

```diff
@@ -778,7 +775,7 @@ public static long toLong(byte[] bytes, int offset, final int length) {
     if (length != SIZEOF_LONG || offset + length > bytes.length) {
       throw explainWrongLengthOrOffset(bytes, offset, length, SIZEOF_LONG);
     }
-    return ConverterHolder.BEST_CONVERTER.toLong(bytes, offset, length);
+    return ConverterHolder.getBestConverter().toLong(bytes, offset, length);
```

Comment:
My expectation here is by the time the code is JITted the JVM will be able to predict ConverterHolder#getBestConverter() is both monomorphic and returns the same value for every invocation. There will be a bit of extra code to typecheck and branch (and de-opt) if the prediction fails, but I believe the branches will have the right prediction hints. The generated code will have the same performance characteristics as previous. If this is a concern, a jmh-based microbenchmark could confirm.